File: 01_CODE/app/src/main/java/com/beigel/leetSpeak_Generator/LeetOption.java
package com.beigel.leetSpeak_Generator;

public class LeetOption {
    private final int mode;
    private final String name;
    private final String description;
    private final int iconResId;
    private final boolean isCustom;
    private final int customIndex;
    private boolean isSelected;
    private boolean isFavorite;

    public LeetOption(int mode, String name, String description, int iconResId,
                      boolean isCustom, int customIndex, boolean isSelected, boolean isFavorite) {
        this.mode = mode;
        this.name = name;
        this.description = description;
        this.iconResId = iconResId;
        this.isCustom = isCustom;
        this.customIndex = customIndex;
        this.isSelected = isSelected;
        this.isFavorite = isFavorite;
    }

    // Getters
    public int getMode() { return mode; }
    public String getName() { return name; }
    public String getDescription() { return description; }
    public int getIconResId() { return iconResId; }
    public boolean isCustom() { return isCustom; }
    public int getCustomIndex() { return customIndex; }
    public boolean isSelected() { return isSelected; }
    public boolean isFavorite() { return isFavorite; }

    // Setters
    public void setSelected(boolean selected) { isSelected = selected; }
    public void setFavorite(boolean favorite) { isFavorite = favorite; }
}

File: 01_CODE/app/src/main/java/com/beigel/leetSpeak_Generator/LeetOption.kt
package com.beigel.leetSpeak_Generator

/**
 * Represents a leet translation option in the UI
 * Converted to Kotlin data class with improved immutability and type safety
 */
data class LeetOption(
    val mode: Int,
    val name: String,
    val description: String,
    val iconResId: Int,
    val isCustom: Boolean = false,
    val customIndex: Int = -1,
    var isSelected: Boolean = false,
    var isFavorite: Boolean = false
) {

    /**
     * Creates a copy with updated selection state
     */
    fun withSelection(selected: Boolean): LeetOption =
        copy(isSelected = selected)

    /**
     * Creates a copy with updated favorite state
     */
    fun withFavorite(favorite: Boolean): LeetOption =
        copy(isFavorite = favorite)

    /**
     * Toggles the favorite state and returns a new instance
     */
    fun toggleFavorite(): LeetOption =
        copy(isFavorite = !isFavorite)

    /**
     * Checks if this option represents a built-in leet mode
     */
    val isBuiltIn: Boolean
        get() = !isCustom

    /**
     * Gets a display-friendly description including state indicators
     */
    val displayDescription: String
        get() = buildString {
            append(description)
            if (isFavorite) append(" ⭐")
            if (isSelected) append(" ✓")
        }

    companion object {
        /**
         * Creates a Simple Leet option
         */
        fun createSimple(isSelected: Boolean = false, isFavorite: Boolean = false): LeetOption =
            LeetOption(
                mode = ProfileManager.MODE_SIMPLE,
                name = "Simple Leet",
                description = "Einfache Leetspeak-Übersetzung",
                iconResId = R.drawable.ic_simple_mode,
                isCustom = false,
                customIndex = -1,
                isSelected = isSelected,
                isFavorite = isFavorite
            )

        /**
         * Creates an Extended Leet option
         */
        fun createExtended(isSelected: Boolean = false, isFavorite: Boolean = false): LeetOption =
            LeetOption(
                mode = ProfileManager.MODE_EXTENDED,
                name = "Extended Leet",
                description = "Erweiterte Leetspeak-Übersetzung",
                iconResId = R.drawable.ic_extended_mode,
                isCustom = false,
                customIndex = -1,
                isSelected = isSelected,
                isFavorite = isFavorite
            )

        /**
         * Creates a Custom Leet option from a profile
         */
        fun createCustom(
            profile: CustomProfile,
            customIndex: Int,
            isSelected: Boolean = false,
            isFavorite: Boolean = false
        ): LeetOption =
            LeetOption(
                mode = ProfileManager.MODE_CUSTOM,
                name = profile.name,
                description = "Benutzerdefiniertes Leet",
                iconResId = profile.iconResId,
                isCustom = true,
                customIndex = customIndex,
                isSelected = isSelected,
                isFavorite = isFavorite
            )
    }
}